pkg/synthetictests/disruption.go

```diff
@@ -48,6 +49,19 @@ func testServerAvailability(
 	connType := monitorapi.DisruptionConnectionTypeFrom(locatorParts)
 	backendName := fmt.Sprintf("%s-%s-connections", disruptionName, connType)
 	if jobType == nil {
+		// check for MicroShift cluster
+		kubeClient, err := framework.LoadClientset(true)
+		if err != nil {
+			panic(err)
+		}
+		isMicroShift, err := exutil.IsMicroShiftCluster(kubeClient)
```

Comment:
Instead of giving empty list of junit test cases, we should ask ourselves whether we need to run `testServerAvailability` at all. Checking the code:
```sh
$ grep -rn "testServerAvailability"
pkg/synthetictests/disruption.go:37:func testServerAvailability(
pkg/synthetictests/disruption.go:190:		ret = append(ret, testServerAvailability("sig-network-edge", locator, allDisruptionEventsIntervals, jobRunDuration, jobType)...)
pkg/synthetictests/disruption.go:209:		ret = append(ret, testServerAvailability("sig-trt", locator, allDisruptionEventsIntervals, jobRunDuration, jobType)...)
```
which leads to https://github.com/openshift/origin/blob/master/pkg/synthetictests/event_junits.go#L38-L39:
```go
	tests = append(tests, TestAllIngressBackendsForDisruption(events, duration, jobType)...)
	tests = append(tests, TestExternalBackendsForDisruption(events, duration, jobType)...)
```

Question here is whether it is safe to skip `TestAllIngressBackendsForDisruption` and `TestExternalBackendsForDisruption`? If the answer is not, then instead of checking whether an installation is MicroShift one should check whether the ingress/externalbackends are present.

Reply:
All these disruptions depend upon `JobType` and for MicroShift this will always be nil. So, I think we don't want to run any of these disruption invariants for MicroShift. 

And to make this happen, I see the possible way is either call `IsMicroShiftCluster()` inside `testServerAvailability` and return an empty list of junit test cases (which I've done above), or 

not to append the following to `tests` based on the same `IsMicroShiftCluster()` check inside `event_junits.go` 
```go
	tests = append(tests, TestAllAPIBackendsForDisruption(events, duration, jobType)...)
	tests = append(tests, TestAllIngressBackendsForDisruption(events, duration, jobType)...)
	tests = append(tests, TestExternalBackendsForDisruption(events, duration, jobType)...)
```

/cc @pacevedom @dhellmann  let me know your thoughts.